src/lib.rs

```
@@ -95,32 +98,80 @@ use std::fmt::{self, Debug, Display, Formatter};
 /// - The `data` must be in the OpenType font format.
 /// - The `index` is only relevant if the data contains a font collection
 ///   (`.ttc` or `.otc` file). Otherwise, it should be 0.
-pub fn subset(data: &[u8], index: u32, mapper: &GlyphRemapper) -> Result<Vec<u8>> {
+///
+/// Important note: If the `variable_fonts` feature is disabled, CFF2 fonts are
+/// not supported at all and will result in an error. If the features is enabled,
+/// CFF2 fonts will be converted into a TrueType font.
+pub fn subset(
+    data: &[u8],
+    index: u32,
+    variation_coordinates: &[(String, f32)],
```

considering that this would just be silently ignored if the feature is disabled, I'd suggest introducing a second cfg-guarded function `subset_with_variations` instead. then we also don't have a breaking change.

---

The question is also whether to support CFF2 in the non-variation subset function then. I'm not sure changing behaviour subtly like this based on a feature is good practice considering that features are unified across the dependency tree and the behaviour of one dependency could then be altered by a peer dependency. Essentially, it violates the principle that features should be additive rather than modificative.

---

But isn't it additive already? CFF2 will be rejected if the feature is disabled, but work when it's enabled.

Will change the function signature thing, though.

---

Depends on the definition of additive I guess. If additive means "add extra API surface, but do not change the behaviour of existing API surface", then it's not additive.

---

I’m still not sure I completely understand, but is it okay the way it is now?

---

Yes, it's okay now. The problem would only be if CFF2 support was silently added to `fn subset` based on the feature.